crates/wasi-http/src/lib.rs

```diff
@@ -1,27 +1,132 @@
 use crate::component_impl::add_component_to_linker;
-pub use crate::r#struct::WasiHttp;
+pub use crate::http_impl::WasiHttpViewExt;
+pub use crate::r#struct::{WasiHttpCtx, WasiHttpView};
+use core::fmt::Formatter;
+use std::fmt::{self, Display};
 
-wasmtime::component::bindgen!({ path: "wasi-http/wit", world: "proxy"});
+wasmtime::component::bindgen!({
+    path: "wasi-http/wit",
+    world: "proxy",
+    with: {
+        "wasi:io/streams": wasmtime_wasi::preview2::bindings::io::streams,
+        "wasi:poll/poll": wasmtime_wasi::preview2::bindings::poll::poll,
+    }
+});
 
 pub mod component_impl;
 pub mod http_impl;
-pub mod streams_impl;
 pub mod r#struct;
 pub mod types_impl;
 
-pub fn add_to_component_linker<T>(
-    linker: &mut wasmtime::component::Linker<T>,
-    get_cx: impl Fn(&mut T) -> &mut WasiHttp + Send + Sync + Copy + 'static,
-) -> anyhow::Result<()> {
-    crate::wasi::http::outgoing_handler::add_to_linker(linker, get_cx)?;
-    crate::wasi::http::types::add_to_linker(linker, get_cx)?;
-    crate::wasi::io::streams::add_to_linker(linker, get_cx)?;
+pub fn add_to_component_linker<T>(linker: &mut wasmtime::component::Linker<T>) -> anyhow::Result<()>
+where
+    T: WasiHttpView
+        + WasiHttpViewExt
+        + crate::wasi::http::outgoing_handler::Host
+        + crate::wasi::http::types::Host,
+{
+    crate::wasi::http::outgoing_handler::add_to_linker(linker, |t| t)?;
+    crate::wasi::http::types::add_to_linker(linker, |t| t)?;
     Ok(())
 }
 
-pub fn add_to_linker<T>(
-    linker: &mut wasmtime::Linker<T>,
-    get_cx: impl Fn(&mut T) -> &mut WasiHttp + Send + Sync + Copy + 'static,
-) -> anyhow::Result<()> {
-    add_component_to_linker(linker, get_cx)
+pub fn add_to_linker<T>(linker: &mut wasmtime::Linker<T>) -> anyhow::Result<()>
```

Comment:
Why remove the closure argument here? (That's what's causing the CI build failures currently.)

Reply:
it will make more sense when using the `preview2::WasiCtx` as laid out in [this pr](https://github.com/bytecodealliance/wasmtime/pull/6836). the wasmtime cli tests will be fixed then